ember-core/shared/src/main/scala/org/http4s/ember/core/Encoder.scala

```diff
@@ -140,7 +140,9 @@ private[ember] object Encoder {
         }
 
         if (
-          !appliedContentLength && req.body == EmptyBody && !NoPayloadMethods.contains(req.method)
+          !appliedContentLength && req.entity == Entity.Empty && !NoPayloadMethods.contains(
+            req.method
+          )
```

Comment:
Nice :)

Reply:
Not really ._.

Reply:
Yeah the formatting got a bit ugly. But I think if we can land https://github.com/http4s/http4s/pull/6762 then we can make this a lot nicer and also more efficient.